.github/workflows/nodejs.yml

```diff
@@ -1,16 +1,13 @@
 name: Node CI
 
-on: [push, pull_request]
```

Comment:
The tests must run both on push and pull request on every branch. After you've tested properly, the deployment should only happen on the `develop` and `master` branches with varying configs.